--- a/net/minecraft/world/gen/settings/DimensionGeneratorSettings.java
+++ b/net/minecraft/world/gen/settings/DimensionGeneratorSettings.java
@@ -210,6 +210,8 @@
       Registry<Biome> registry = p_242753_0_.func_243612_b(Registry.field_239720_u_);
       Registry<DimensionSettings> registry1 = p_242753_0_.func_243612_b(Registry.field_243549_ar);
       SimpleRegistry<Dimension> simpleregistry = DimensionType.func_242718_a(registry2, registry, registry1, i);
+      net.minecraftforge.common.world.ForgeWorldType type = net.minecraftforge.registries.ForgeRegistries.WORLD_TYPES.getValue(new net.minecraft.util.ResourceLocation(s4));
+      if (type != null) return new DimensionGeneratorSettings(i, flag, false, DimensionGeneratorSettings.func_242749_a(registry2, simpleregistry, type.createChunkGenerator(registry, registry1, i, s)));
       byte b0 = -1;
       switch(s4.hashCode()) {
       case -1100099890:
